src/navigators/createMaterialTopTabNavigator.js

```diff
@@ -113,18 +133,109 @@ class TabView extends React.PureComponent<Props> {
 
   _renderPanPager = props => <TabViewPagerPan {...props} />;
 
-  _renderScene = ({ route, focused }) => {
-    const { renderScene, animationEnabled, swipeEnabled } = this.props;
+  componentWillMount() {
+    const { animationEnabled } = this.props;
 
-    if (animationEnabled === false && swipeEnabled === false) {
-      return (
-        <ResourceSavingScene isFocused={focused}>
-          {renderScene({ route })}
-        </ResourceSavingScene>
+    // We rely on action listener to animate from the current index to the next one, no need to listen without animation
+    if (animationEnabled) {
+      this._actionListener = this.props.navigation.addListener(
+        'action',
+        this._onAction
       );
     }
+  }
+
+  componentWillUnmount() {
+    if (this._actionListener) {
+      this._actionListener.remove();
+    }
+
+    if (this._transitionTimeout) {
+      clearTimeout(this._transitionTimeout);
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
`componentWillRecieveProps` is deprecated. We'll need to switch to `static getDerivedStateFromProps` with `react-lifecycle-compat`.